crates/re_types_builder/src/codegen/rust/serializer.rs

```diff
@@ -410,6 +402,15 @@ fn quote_arrow_field_serializer(
     };
     let inner_is_arrow_transparent = inner_obj.map_or(false, |obj| obj.datatype.is_none());
 
+    assert!(
+        !is_nullable || validity_bitmap.is_some(),
+        "Nullable fields must have a validity bitmap.
+        Note though that if a field is not nullable, we may still use a validity bitmap."
```

Comment:
use it for what?

Reply:
The short answer is that I don't know all that well. Wasn't able to figure it out properly so far.

There's a set of workaround further down that Clement added at some point that need a validity map for inner arrays where we otherwise don't use validity maps. But the main thing that kept me from making nullable==false -> no validity map (and it's the first thing I tried) is the ripple effects it has on code that currently always assumed a validity map to be present. Today, all code always pretends there is one (often doing things like `.map(Some)` to enforce it on the data) in order to share the same codepath.

Reply:
I'm starting to realize what I got wrong here 💡 : `is_nullable` is about the field itself being nullable, so to speak the outer part of `Option<Vec<Option<Thing>>>`. The inner part is what we handle with the validity_bitmap and since we're implementing `to_arrow_opt` we always need it.
So now with the unions we take care about `Option<Thing>` with the special enum variant, so this is the one and only case where we don't need a validity bitmap. _Additionally_ we also don't support validity bitmaps in the inner arrays of arrays.

I'll try to clear/clean this up a bit.

(whether we actually need to take in vecs of optional things like ever is a different story entirely....)

Reply:
75733213b0 clears this up through removing `is_nullable` which wasn't actually used (wow..) and adding docs :)

Reply:
and further improved things in [fa06520](https://github.com/rerun-io/rerun/pull/6236/commits/fa0652058ed700c646e508ede2619a80f279f623)
making use of the little new found understanding of the code I have to cull out some strange `.map(Some)` which are `unwrap_or_default`'ed right away again.